app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/ReviewRegistrationController.scala

```diff
@@ -33,24 +38,38 @@ class ReviewRegistrationController @Inject() (
   journeyAction: JourneyAction,
   mcc: MessagesControllerComponents,
   incorpIdConnector: IncorpIdConnector,
+  override val registrationConnector: RegistrationConnector,
   page: review_registration_page
 )(implicit ec: ExecutionContext)
-    extends FrontendController(mcc) with I18nSupport {
+    extends FrontendController(mcc) with Cacheable with I18nSupport {
 
   def displayPage(): Action[AnyContent] =
     (authenticate andThen journeyAction).async { implicit request =>
-      request.registration.incorpJourneyId match {
-        case Some(journeyId) =>
-          incorpIdConnector.getDetails(journeyId).map(
-            details => Ok(page(request.registration, details))
-          )
-        case _ => Future(Redirect(routes.RegistrationController.displayPage()))
-      }
+      if (request.registration.isCheckAndSubmitReady)
+        request.registration.incorpJourneyId match {
+          case Some(journeyId) =>
+            for {
+              incorporationDetails <- incorpIdConnector.getDetails(journeyId)
+              _                    <- updateRegistration(hasReviewedRegistration = true)
```

Comment:
Curious as to why this is being updated upon loading this page.  Or is that the point of the `hasReviewedRegistration` flag - to prove that the user _has_ seen this page.

Reply:
Yes exactly it.